dex/runner.go

```diff
@@ -105,6 +105,9 @@ func (c *ConnectionMaster) Connect(ctx context.Context) (err error) {
 	c.mtx.Lock()
 	c.wg, err = c.connector.Connect(c.ctx)
 	c.mtx.Unlock()
+	if err != nil {
+		c.cancel() // otherwise On() says true
+	}
```

Comment:
Note sure how we didn't have trouble with this before.  This allows a wallet whose `ConnectionMaster` has returned a non-nil error from `Connect` to look like we expect:

![image](https://user-images.githubusercontent.com/9373513/153093795-5a69d41a-e494-42bc-9fdd-2d7b6487f7e9.png)

Otherwise you just see the syncing circle-arrow icon and no "Connect" button